Upgrade Table authoring to Angular component

geoffreykwan opened the pull request:
Make sure Table authoring still works.

Closes #2902

hirokiterashima commented on src/main/webapp/wise5/components/table/table-authoring/table-authoring.component.ts:
```diff
@@ -0,0 +1,365 @@
+  insertColumn(x: number): void {
+    const tableData = this.authoringComponentContent.tableData;
+    const numRows = this.authoringComponentContent.numRows;
+    for (let r = 0; r < numRows; r++) {
+      const tempRow = tableData[r];
+      if (tempRow != null) {
+        const newCell = this.createEmptyCell();
+        tempRow.splice(x, 0, newCell);
+      }
+    }
```
Is this null check necessary?

hirokiterashima commented on src/main/webapp/wise5/components/table/table-authoring/table-authoring.component.ts:
```diff
@@ -0,0 +1,365 @@
+    if (confirm($localize`Are you sure you want to delete this column?`)) {
+      const tableData = this.authoringComponentContent.tableData;
+      const numRows = this.authoringComponentContent.numRows;
+      for (let r = 0; r < numRows; r++) {
+        const tempRow = tableData[r];
+        if (tempRow != null) {
+          tempRow.splice(x, 1);
+        }
+      }
```
Is this null check necessary?

hirokiterashima reviewed (changes requested):
Functionality looks good. I cleaned up the code a little bit. Also found a part of the code where I wasn't sure if the null-check is necessary and added comments inline. Can you take a look?

geoffreykwan commented:
@hirokiterashima Ok I removed those null checks because they didn't seem to be needed. I also set the min value for inputs to 1 and changed some variable names.

hirokiterashima reviewed (approved):
Looks good.
